python/hsfs/core/vector_server.py

```diff
@@ -82,17 +84,29 @@ def __init__(
         )
         self._transformation_functions = None
         self._required_serving_keys = None
+        self._async_pool = None
 
     def init_serving(
-        self, entity, batch, external, inference_helper_columns=False, options=None
+        self,
+        entity,
+        batch,
+        external,
+        inference_helper_columns=False,
+        options=None,
+        parallel=False,
```

Comment:
parallel should be a number, not boolean flag.
And it should be 'none'. By default, we should have it as the number of parallel threads (MySQL Connections) - the number of prepared statements.

Most often, a hsfs client is used by one model and every call will have the same number of prepared statements. We can set the minsize and maxsize for aiomysql.create_pool(..) to be the same value. That will ensure it maximizes performance.

Reply:
Removed the parallel flag and the sqlalchemy engine entirely during get_feature_vector(s)